_posts/en/newsletters/2025-06-13-newsletter.md

```diff
@@ -0,0 +1,156 @@
+---
+title: 'Bitcoin Optech Newsletter #358'
+permalink: /en/newsletters/2025/06/13/
+name: 2025-06-13-newsletter
+slug: 2025-06-13-newsletter
+type: newsletter
+layout: newsletter
+lang: en
+---
+This week's newsletter FIXME:harding
+
+## News
+
+- **Calculating the selfish mining danger threshold:** Antoine Poinsot
+  [posted][poinsot selfish] to Delving Bitcoin an expansion of the math
+  from the 2013 [paper][es selfish] that gave the [selfish mining
+  attack][topic selfish mining] it's name (although the attack was
+  [previously described][bytecoin selfish] in 2010).  He also provided a
+  simplified mining and block relay
+  [simulator][darosior/miningsimulation] that allows experimenting with
+  the attack.  He focuses on reproducing one of the conclusions of the
+  2013 paper: that a dishonest miner (or cartel of well-connected
+  miners) controlling 33% of total network hashrate, with no additional
+  advantages, can become marginally more profitable on a long term basis
+  than the miners controlling the majority 67% of hashrate.  This is
+  achieved by the 33% miner selectively delaying announcement of some of
+  the new blocks it finds.  As the dishonest miner's hashrate increases
+  above 33%, it becomes even more profitable until it exceeds 50%
+  hashrate and is able to prevent its competitors from keeping any new
+  blocks on the best blockchain.
+
+  We did not carefully review Poinsot's post, but his approach looked
+  sound to us and we would recommend it to anyone interested in
+  validating the math or better understanding it.
+
+- **Relay censorship resistance through top mempool set reconciliation:**
+  Peter Todd [posted][todd feerec] to the Bitcoin-Dev mailing list about
+  a mechanism that would allow nodes to drop peers that are filtering
+  high-feerate transactions.  The mechanism depends on [cluster
+  mempool][topic cluster mempool] and a set reconciliation mechanism
+  such as that used by [erlay][topic erlay].  A node would use cluster
+  mempool to calculate its most profitable set of unconfirmed
+  transactions that could fit in (for example) 8,000,000 weight units (a
+  max of 8 MB).  Each of the node's peers would also calculate their top
+  8 MWU of unconfirmed transactions.  Using a highly efficient
+  algorithm, such as [minisketch][topic minisketch], the node would
+  reconcile its set of transactions with each of its peers.  In doing
+  so, it would learn exactly what transactions each peer has in the top
```

Comment:
```suggestion
  so, it would learn exactly what transactions each peer has at the top
```

Reply:
Leaving as-is.  I think "in" makes more sense when we're talking about set reconciliation, as items are "in" a set (or not) rather than "at" a set.